### PR TITLE
keepassxc: update to 2.5.4

### PR DESCRIPTION
**Pull request by augiedoggie**

### CODE REVIEW

**Comment by humdingerb** [comment]
A few icons reverted to somewhat ugly ones... see https://oshi.at/sJkbeZ/old-new-icons.png

File: app-admin/keepassxc/keepassxc-2.5.4.recipe [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+INSTALL()
+{
+#	TODO: adjust path for /boot/home/.config/keepassxc/keepassxc.ini
+#	TODO: replace icon with the one from keepassxc.svgz
+#	TODO: replace toolbar icons with https://github.com/HaikuArchives/ZumisIcons/tree/master/myhaiku/btoolbar
+	cd build
+	make install
+
+	appIconsDir=$dataDir/keepassxc/icons/application
```
Comment (humdingerb) [inline app-admin/keepassxc/keepassxc-2.5.4.recipe:113]: I suppose this line can be commented out as well. $appIconsDir appears to be never used.

File: app-admin/keepassxc/keepassxc-2.5.4.recipe [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		-DWITH_TESTS=off \
+		-DCMAKE_CXX_FLAGS="-D_GNU_SOURCE"
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+#	TODO: adjust path for /boot/home/.config/keepassxc/keepassxc.ini
+#	TODO: replace icon with the one from keepassxc.svgz
```
[the remark this answers is not in the document]
Comment (augiedoggie) [reply app-admin/keepassxc/keepassxc-2.5.4.recipe:108]: I thought about removing that.  In most cases the settings file goes to the correct location but could be improved.  I've had cases where the XDG environment variables weren't set and the settings ended up in ~/.config/keepassxc.ini

**Comment by augiedoggie** [comment]
> A few icons reverted to somewhat ugly ones... see https://oshi.at/sJkbeZ/old-new-icons.png

I hadn't noticed that until now.  I'm not sure if that's controlled by the app or Qt

**Comment by humdingerb** [comment]
I believe these come from Qt. At least these ugly ones and the edit and delete entry icons are not in keepass' share/icons folder. No idea how to fix though...

**Comment by augiedoggie** [comment]
I've made the cleanups to the TODOs and squashed the commits.  I'm not sure why the app is preferring to use standard Qt icons instead of the bundled ones.  I don't have the time/energy to look into it at the moment.

**Comment by humdingerb** [comment]
Let's see if @threedeyes  or @diversys have some input on the icon issue. Guys?

**Comment by diversys** [comment]
Did you forget to upload a patchset? I can only see the recipe. BTW, did you git mv the recipe or just added a new one?

**Comment by augiedoggie** [comment]
> Did you forget to upload a patchset? I can only see the recipe. BTW, did you git mv the recipe or just added a new one?

It appeared as though the patches were already submitted upstream.  It's just a copy of the recipe, I didn't 'git mv' it

**Comment by Begasus** [comment]
Patches were upstreamed, thanks to @diversys :) https://github.com/keepassxreboot/keepassxc/pull/2696
@augiedoggie could you drop the older recipe and patchset?

File: app-admin/keepassxc/keepassxc-2.5.4.recipe [hunk 3]
```diff
@@ -25,13 +25,13 @@ COPYRIGHT="
...
+	2016-2020, KeePassXC Team
 	"
 LICENSE="GNU GPL v2"
-REVISION="2"
+REVISION="1"
 SOURCE_URI="https://github.com/keepassxreboot/keepassxc/archive/$portVersion.tar.gz"
-CHECKSUM_SHA256="3610eccb06b04f901d5f5fbdbc4bd474d56a8e35f23c4a7873a844e44ba762b4"
-PATCHES="keepassxc-$portVersion.patchset"
+CHECKSUM_SHA256="ebb27af867548f765d2b3aeb3937ad34f429159f7ce12af7a5db6a744997c5aa"
+#PATCHES="keepassxc-$portVersion.patchset"
```
Comment (Begasus) [inline app-admin/keepassxc/keepassxc-2.5.4.recipe:34]: You can remove the line if it's not needed anymore :)
```diff
- #PATCHES="keepassxc-$portVersion.patchset"
```

**Pull request merged**